compiler/rustc_span/src/caching_source_map_view.rs

```diff
@@ -123,27 +123,25 @@ impl<'sm> CachingSourceMapView<'sm> {
 
         if lo_cache_idx != -1 && hi_cache_idx != -1 {
             // Cache hit for span lo and hi. Check if they belong to the same file.
-            let result = {
-                let lo = &self.line_cache[lo_cache_idx as usize];
-                let hi = &self.line_cache[hi_cache_idx as usize];
+            let lo_file_index = self.line_cache[lo_cache_idx as usize].file_index;
```

Comment:
```suggestion
            let lo = &self.line_cache[lo_cache_idx as usize];
```
Not sure why this change.
`lo.file_index` -> `lo_file_index` is not a cleanup, and `self.line_cache[lo_cache_idx as usize]` is used below a couple of times.

Reply:
`update` takes a `&mut`, so we need to either (1) separate the `lo_cache_idx == hi_cache_idx` and use `pick2_mut`, or (2) make use a `Cell` inside `CacheEntry`.

Reply:
Ah, I see.